src/components/Dialog/__tests__/Dialog.interactions.ts

```diff
@@ -0,0 +1,59 @@
+import { resetFocus } from "../../../__tests__/interactions-helper";
+import { expect } from "@storybook/jest";
+import { BoundFunctions, Screen } from "@testing-library/react";
+import {
+  CONTEXT_MENU_DIALOG,
+  CLICK_OUTSIDE_DIALOG,
+  HIDE_TRIGGERS_CONTAINER,
+  CLICK_OUTSIDE_DIALOG_BUTTON
+} from "./DialogDataTestIds";
+import { getByTestId, interactionSuite } from "../../../tests/interactions-utils";
+import { queries, userEvent, fireEvent, waitFor } from "@storybook/testing-library";
+
+const isDialogHiddenAfterClickOutside = createTestIfDialogHiddenAfterTrigger(CLICK_OUTSIDE_DIALOG, () =>
+  userEvent.click(getDialogContainer())
+);
+const isDialogHiddenAfterContextMenu = createTestIfDialogHiddenAfterTrigger(CONTEXT_MENU_DIALOG, () =>
+  fireEvent.contextMenu(getDialogContainer())
+);
+
+export const closeTriggersInteractionSuite = interactionSuite({
```

Comment:
in general, the test boilerplate looks really complex, so I think we should think about how to make it more simple

Reply:
Agree, adding a backlog task / reaserch task

Reply:
https://monday.monday.com/boards/3532714909/pulses/4587109938